spag/template.py

```diff
@@ -93,26 +118,71 @@ def _substitute_braces(s, withs, body_type):
                 .format(original))
 
     for item in items:
-        # items without a dot are specfied using `--with item=val`
-        if '.' not in item and item in withs:
+        if item.startswith('['):
+            try:
+                return _lookup_item_from_environment(item)
+            except common.ToughNoodles:
+                pass
+        elif '.' in item:
+            try:
+                return _lookup_item_from_request(item, body_type)
+            except common.ToughNoodles:
+                pass
+        elif item in withs:
             return withs[item]
-        elif '.' not in item:
-            continue
-
-        try:
-            return _lookup_item_with_dot(item, body_type)
-        except common.ToughNoodles:
-            pass
 
+    # haven't found a value yet. try returning the default.
     if default is not None:
         return default
 
     raise common.ToughNoodles("Failed to substitute for {0}".format(original))
 
-def _lookup_item_with_dot(item, body_type):
+def _dict_path_lookup(data, path):
+    """If path == "a.b.c", return data['a']['b']['c']. Raise ToughNoodles if
+    the value is not found.
+    """
+    path = path.strip('.').split('.')
+    try:
+        for key in path:
+            # TODO: do we care about this?
+            # this won't work if we have a list somewhere along the way:
+            #   data = {a: [{b: 1}, {b: 2}]}
```

Comment:
This could happen in a JSON response, called from L202. 
```json
{
  "name": "pglbutt.org.",
  "maintainers": [
    {
       "name": "pglbutt",
       "motto": "I'll write my own"
    },
    {
      "name": "timsim",
      "motto": "let's go to luby's"
    }
  ]
}
```

We'll have to worry about that.